server/main_server/src/main.cpp

```diff
@@ -1450,6 +1484,248 @@ Environment Variables:
+                    std::pair{irods::KW_CFG_ACCESS_TIME_RESOLUTION_IN_SECONDS, &resolution_in_seconds}
+                };
+
+                for (auto&& opt : opt_names) {
+                    std::strcpy(input.option_name, opt.first);
+
+                    // NOLINTNEXTLINE(cppcoreguidelines-owning-memory,cppcoreguidelines-no-malloc)
+                    irods::at_scope_exit free_output{[&output] { std::free(output); }};
+
+                    if (const auto ec = rc_get_grid_configuration_value(static_cast<RcComm*>(conn), &input, &output); ec < 0) {
+                        log_server::error("{}: Could not retrieve grid configuration value from catalog [error_code={}, namespace=access_time, option_name={}].",
+                                __func__, ec, opt.first);
+
+                        // We break out of the for-loop instead of returning to allow more error
+                        // information to be logged and so that the consumer branch is closer in behavior
+                        // to the provider branch.
+                        break;
+                    }
+
+                    *opt.second = output->option_value;
+                }
+            }
+
+            //
+            // At this point, we expect to have all access time configuration values available.
+            //
+
+            if (!queue_name_prefix) {
+                log_server::error("{}: Failed to initialize access time queue: Is [access_time.queue_name_prefix] defined in R_GRID_CONFIGURATION?", __func__);
+                return false;
+            }
+
+            if (!queue_size) {
+                log_server::error("{}: Failed to initialize access time queue: Is [access_time.queue_size] defined in R_GRID_CONFIGURATION?", __func__);
+                return false;
+            }
+
+            if (!batch_size) {
+                log_server::error("{}: Failed to initialize access time queue: Is [access_time.batch_size] defined in R_GRID_CONFIGURATION?", __func__);
+                return false;
+            }
+
+            if (!resolution_in_seconds) {
+                log_server::error("{}: Failed to initialize access time queue: Is [access_time.resolution_in_seconds] defined in R_GRID_CONFIGURATION?", __func__);
+                return false;
+            }
+
+            irods::access_time_queue::init(*queue_name_prefix, std::stoull(*queue_size));
+
+            // Store configuration values in global variables for later use. 
+            g_atime_batch_size = std::stoull(*batch_size);
+            g_atime_resolution_in_seconds = *std::move(resolution_in_seconds);
+
+            return true;
+        }
+        catch (const irods::exception& e) {
+            log_server::error("{}: Failed to initialize access time queue: {}", __func__, e.client_display_what());
+        }
+        catch (const std::exception& e) {
+            log_server::error("{}: Failed to initialize access time queue: {}", __func__, e.what());
+        }
```

Comment:
Is there any value in handling `nanodbc` exceptions specifically?

Reply:
Depends on if we should do something different from the std::exception case.

Reply:
We probably don't need it, then. I can't think of anything unless the exception structure has some extra info that could be useful.

Reply:
The std::exception will include the database error codes, so everything is there.

Resolving.

Reply:
Just to be clear ... when the nanodbc exception is caught by the std::exception block, the database error codes will be included in the log message.